src/axolotl/utils/models.py

```diff
@@ -367,11 +366,10 @@ def load_model(
         model = prepare_model_for_kbit_training(
             model, use_gradient_checkpointing=cfg.gradient_checkpointing
         )
-        needs_fa2_dtype = True
 
     # LlamaRMSNorm layers are in fp32 after kbit_training or full finetune, so we need to
     # convert them back to fp16/bf16 for flash-attn compatibility.
-    if needs_fa2_dtype and (cfg.flash_attention and cfg.is_llama_derived_model):
+    if cfg.flash_attention and cfg.is_llama_derived_model:
```

Comment:
I think it's should be when it's a lora OR flash attention. (still need to add back the removed lines above)
```suggestion
    if needs_fa2_dtype or (cfg.flash_attention and cfg.is_llama_derived_model):
```